model/task_lifecycle.go

```diff
@@ -1451,28 +1451,25 @@ func ClearAndResetStrandedTask(h *host.Host) error {
 		return errors.WithStack(MarkEnd(t, evergreen.MonitorPackage, time.Now(), &t.Details, false))
 	}
 
+	// Only restart failed tasks for stranded tasks.
+	if err := t.SetRestartFailed(); err != nil {
```

Comment:
This isn't right since in ResetTaskOrDisplayTask we still call SetResetWhenFinished (that's the case where we should be setting failed i think)

Reply:
ResetTaskOrDisplayTask is only called in the reset route and here so I created a new ticket to add it to the route